src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureBlobStorageOptions.cs

```diff
@@ -54,9 +65,18 @@ public AzureBlobStorageOptionsValidator(AzureBlobStorageOptions options, string
 
         public void ValidateConfiguration()
         {
-            if (!CloudStorageAccount.TryParse(this.options.ConnectionString, out var ignore))
-                throw new OrleansConfigurationException(
-                    $"Configuration for AzureBlobStorageOptions {name} is invalid. {nameof(this.options.ConnectionString)} is not valid.");
+            if (this.options.ServiceUri == null)
```

Comment:
Is this logic correct? If ServiceUri is set then we perform no validation on the TokenCredential or the ConnectionString?

Finally, does TokenCredential need to be marked as `[RedactConnectionString]`

Otherwise, LGTM and we can merge.

Reply:
I think the connection string is still required. The only thing is that if you have the credentials set, you don't need to have the access keys on the connection string.

Reply:
It's missing an `else` to check that `TokenCredential` is not null. 
`ConnectionString` should not be used with AAD auth.

Reply:
I'm not sure redact is needed here, the credential's `ToString` would not print any secrets.

Reply:
NVM, the connection string is used for SQL with managed identity. Ignore that...